distributed/scheduler.py

```diff
@@ -4138,6 +4193,52 @@ def decide_worker(self, ts):
 
         return worker
 
+    def transition_waiting_speculative(self, key):
+        try:
+
+            ts = self.tasks[key]
+
+            if self.validate:
+                # All dependencies are on the same worker
+                assert len({dts.processing_on for dts in ts.dependencies}) == 1
```

Comment:
We probably also want that at least one of the dependencies is in a "processing" state